### PR TITLE
Add NekCEM package to spack

### PR DESCRIPTION
**Pull request by thilinarmtb**
### Current status

User can use the `makenek` script to build NekCEM cases.

### TODO
- [x] Figure out how to link `blas` correctly

### CODE REVIEW

File: var/spack/repos/builtin/packages/nekcem/package.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        with working_dir(binDir):
+            filter_file(r'^FC\s*=.*', 'FC=\"' + FC + '\"', 'makenek')
+            filter_file(r'^CC\s*=.*', 'CC=\"' + CC + '\"', 'makenek')
+            filter_file(r'^NEK\s*=.*', 'NEK=\"' + prefix.bin.NekCEM +
+                        '\"', 'makenek')
+
+            blasLapack = spec['blas'].libs + spec['lapack'].libs
+            ldFlags = blasLapack.ld_flags
+#            filter_file(r'^EXTRALDFLAGS\s*=.*', 'EXTRALDFLAGS=\"' + str(ldFlags) +
+#                        '\"', 'makenek')
```
Comment (thilinarmtb) [inline var/spack/repos/builtin/packages/nekcem/package.py]: I will remove these comments. What is the correct way to link `blas` libraries?
Do we need to get them from `spec` as in line 74 and 75?

File: var/spack/repos/builtin/packages/nekcem/package.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            FC = spec['mpi'].mpif77
+            CC = spec['mpi'].mpicc
+
+        with working_dir(binDir):
+            filter_file(r'^FC\s*=.*', 'FC=\"' + FC + '\"', 'makenek')
+            filter_file(r'^CC\s*=.*', 'CC=\"' + CC + '\"', 'makenek')
+            filter_file(r'^NEK\s*=.*', 'NEK=\"' + prefix.bin.NekCEM +
+                        '\"', 'makenek')
+
+            blasLapack = spec['blas'].libs + spec['lapack'].libs
```
Comment (davydden) [inline var/spack/repos/builtin/packages/nekcem/package.py]: See http://spack.readthedocs.io/en/latest/packaging_guide.html#blas-lapack-and-scalapack-libraries

Lapack should be first in the sum above

Comment (thilinarmtb) [reply var/spack/repos/builtin/packages/nekcem/package.py]: Thanks @davydden !!

**Comment by thilinarmtb** [comment]
If you guys have no other comments, I think this is ready to be merged as is.

**Pull request merged**